src/librustdoc/visit_ast.rs

```diff
@@ -213,8 +215,36 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
 
     }
 
-    fn resolve_id(&mut self, id: ast::NodeId, renamed: Option<ast::Name>,
+    /// Tries to resolve the target of a `pub use` statement and inlines the
```

Comment:
It may also make sense to inline private imports if we're not stripping private items and the target inherits `doc(hidden)`